File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
         // first reverse 0 to k-1;
         // than reverse k to nums.size();
         // than reverse 0 to nums.size();
        k=k%nums.size();
        reverse(nums.begin(),nums.end());
         reverse(nums.begin()+k, nums.end());
         reverse(nums.begin(),nums.begin()+k);
         

        //  k=k%nums.size();
        // reverse(nums.begin(),nums.end());
        //  reverse(nums.begin(),nums.begin()+k);
        // reverse(nums.begin()+k,nums.end());

    }
};

File: 0189-rotate-array/0189-rotate-array.py
class Solution(object):
    def rotate(self, nums, k):
        l = len(nums)
        my_list = [0]*l
        for i in range(0,len(nums)):
            j = (i+k)%l
            my_list[j] = nums[i]
        nums[:] =  my_list

        